test/unit/client.test.js

```diff
@@ -1019,6 +1058,171 @@ describe('message deletion', () => {
+
+		// Mock the delete API call to simulate server response
+		vi.spyOn(deletingUserClient, 'delete').mockResolvedValue({
+			message: {
+				id: messageId,
+				text: 'Test message',
+				user: deletingUser,
+				deleted_for_me: true,
+				deleted_at: new Date().toISOString(),
+			},
+		});
+	});
+
+	afterEach(() => {
+		vi.resetAllMocks();
+		deletingUserClient.off('message.deleted', deletingUserEventSpy);
+		otherUserClient.off('message.deleted', otherUserEventSpy);
+	});
+
+	it('should send delete_for_me=true to API when deleteForMe is true', async () => {
+		await deletingUserClient.deleteMessage(messageId, false, true);
+
+		expect(deletingUserClient.delete).toHaveBeenCalledWith(
+			`${deletingUserClient.baseURL}/messages/${encodeURIComponent(messageId)}`,
+			{ delete_for_me: true },
+		);
+	});
+
+	it('should handle message.deleted event with deleted_for_me=true for deleting user', () => {
+		const deletedMessage = {
+			id: messageId,
+			text: 'Test message',
+			user: deletingUser,
+			deleted_for_me: true,
+			deleted_at: new Date().toISOString(),
+		};
+
+		// Simulate server sending message.deleted event only to deleting user with deleted_for_me=true
+		const eventForDeletingUser = {
+			type: 'message.deleted',
+			deleted_for_me: true,
+			message: deletedMessage,
+		};
+
+		deletingUserClient.dispatchEvent(eventForDeletingUser);
+
+		expect(deletingUserEventSpy).toHaveBeenCalledWith(eventForDeletingUser);
+		expect(otherUserEventSpy).not.toHaveBeenCalled();
+	});
+
+	it('should not send event to other users when message is deleted for me', () => {
+		const deletedMessage = {
+			id: messageId,
+			text: 'Test message',
+			user: deletingUser,
+			deleted_for_me: true,
+			deleted_at: new Date().toISOString(),
+		};
+
+		// Simulate that only the deleting user receives the event
+		const eventForDeletingUser = {
+			type: 'message.deleted',
+			deleted_for_me: true,
+			message: deletedMessage,
+		};
+
+		deletingUserClient.dispatchEvent(eventForDeletingUser);
+
+		// Other user should not receive any event
+		expect(otherUserEventSpy).not.toHaveBeenCalled();
+		expect(deletingUserEventSpy).toHaveBeenCalledOnce();
+	});
+
+	it('should return deleted_for_me=true when deleting user fetches the message', async () => {
+		const messageWithDeletedForMe = {
+			id: messageId,
+			text: 'Test message',
+			user: deletingUser,
+			deleted_for_me: true,
+			deleted_at: new Date().toISOString(),
+		};
+
+		// Mock getMessage API for deleting user
+		vi.spyOn(deletingUserClient, 'get').mockResolvedValue({
+			message: messageWithDeletedForMe,
+		});
+
+		const result = await deletingUserClient.getMessage(messageId);
+
+		expect(result.message.deleted_for_me).toBe(true);
+		expect(deletingUserClient.get).toHaveBeenCalledWith(
+			`${deletingUserClient.baseURL}/messages/${encodeURIComponent(messageId)}`,
+			{},
+		);
+	});
+
+	it('should return normal message when other user fetches the message', async () => {
```

Comment:
It seems that this test just tests, whether the spy returns the value it was told to return.